### PR TITLE
Add BackingStorageClass and adjust storageCluster template

### PR DESCRIPTION
**Pull request by bindrad**

### CODE REVIEW

File: datafoundation/templates_storagecluster.go [hunk 1]
```diff
@@ -96,32 +101,42 @@ var StorageClusterTemplate = ocsv1.StorageCluster{
...
+		}},
 		StorageProfiles: []ocsv1.StorageProfile{{
-			DeviceClass: "ssd",
-			Name:        "default",
+			DeviceClass: deviceClassName,
+			Name:        storageProfileName,
 			BlockPoolConfiguration: ocsv1.BlockPoolConfigurationSpec{
 				Parameters: map[string]string{
-					"pg_autoscale_mode": "on",
+					"pg_autoscale_mode": "off",
```
[the remark this answers is not in the document]
Comment (bindrad) [reply datafoundation/templates_storagecluster.go:111]: Updated

File: datafoundation/templates_storagecluster.go [hunk 2]
```diff
@@ -29,8 +29,13 @@ const (
 )
 
 var (
-	gp2             = "gp2"
-	volumeModeBlock = corev1.PersistentVolumeBlock
+	gp3MonStorageClassName  = "gp3-csi"
+	volumeModeBlock         = corev1.PersistentVolumeBlock
+	backingStorageClassName = "default"
```
[the remark this answers is not in the document]
Comment (bindrad) [reply datafoundation/templates_storagecluster.go]: Updated

File: datafoundation/templates_storagecluster.go [hunk 3]
```diff
@@ -96,36 +92,48 @@ var StorageClusterTemplate = ocsv1.StorageCluster{
...
 				},
 			},
 		}},
+		DefaultStorageProfile: "default-storage-profile",
 		StorageDeviceSets: []ocsv1.StorageDeviceSet{{
 			Name:        "default",
 			Count:       1,
 			Replica:     3,
-			DeviceClass: "ssd",
+			DeviceClass: "default",
```
Comment (Madhu-1) [inline datafoundation/templates_storagecluster.go:129]: it should be `ssd` this is what we will have in upgraded clusters from ODF 4.11

File: datafoundation/templates_storagecluster.go [hunk 4]
```diff
@@ -96,36 +92,48 @@ var StorageClusterTemplate = ocsv1.StorageCluster{
 			"mon":            GetResourceRequirements("mon"),
 			"crashcollector": GetResourceRequirements("crashcollector"),
 		},
+		BackingStorageClasses: []ocsv1.BackingStorageClass{{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "default-ocs-storage-class",
```
Comment (Madhu-1) [inline datafoundation/templates_storagecluster.go:97]: as used in multiple place define a var/const for it?

File: datafoundation/templates_storagecluster.go [hunk 5]
```diff
@@ -96,36 +92,48 @@ var StorageClusterTemplate = ocsv1.StorageCluster{
...
+			Provisioner: "ebs.csi.aws.com",
+			Parameters: map[string]string{
+				"iops":       "12000",
+				"throughput": "250",
+			},
+		}},
 		StorageProfiles: []ocsv1.StorageProfile{{
-			DeviceClass: "ssd",
-			Name:        "default",
+			DeviceClass: "default",
```
Comment (Madhu-1) [inline datafoundation/templates_storagecluster.go:106]: deviceClass should be `ssd`

**Review by Madhu-1 — APPROVED** [review]

**Review by ezio-auditore — APPROVED** [review]

File: Dockerfile [hunk 6]
```diff
@@ -1,5 +1,5 @@
 # Build the manager binary
-FROM golang:1.17 as builder
+FROM golang:1.19 as builder
```
Comment (nb-ohad) [inline Dockerfile:2]: This is not enough, you need to change the go version in `go.mod` as well

File: datafoundation/templates_storagecluster.go [hunk 7]
```diff
@@ -28,10 +29,7 @@ const (
 	OSDSizeInTiB = 4
 )
 
-var (
-	gp2             = "gp2"
-	volumeModeBlock = corev1.PersistentVolumeBlock
-)
+var backingStorageClassName = "default-ocs-storage-class"
```
Comment (nb-ohad) [inline datafoundation/templates_storagecluster.go:32]: This should become a const, not a var

File: datafoundation/templates_storagecluster.go [hunk 8]
```diff
@@ -96,36 +94,48 @@ var StorageClusterTemplate = ocsv1.StorageCluster{
...
 				},
 			},
 			SharedFilesystemConfiguration: ocsv1.SharedFilesystemConfigurationSpec{
 				Parameters: map[string]string{
-					"pg_autoscale_mode": "on",
-					"pg_num":            "128",
-					"pgp_num":           "128",
+					"pg_autoscale_mode": "off",
+					"pg_num":            "512",
+					"pgp_num":           "512",
```
Comment (nb-ohad) [inline datafoundation/templates_storagecluster.go:122]: @bindrad this seems wrong. In a previous version of ODF, the pg and pgp allocation was a calculation in code. I don't think this setup matches what we had there as we know we had a resource constraint. 

@ezio-auditore Can you help with this as the original calculation (in ocs-operator) was your contribution?

**Review by nb-ohad — CHANGES_REQUESTED** [review]

File: datafoundation/templates_storagecluster.go (hunk 7 above)
Comment (bindrad) [reply datafoundation/templates_storagecluster.go:33]: Updated

File: datafoundation/templates_storagecluster.go (hunk 8 above)
Comment (bindrad) [reply datafoundation/templates_storagecluster.go:123]: I did discuss the configuration with @ezio-auditore  and got the confirmation that this is the one being used in the product.

File: Dockerfile (hunk 6 above)
Comment (nb-ohad) [reply Dockerfile:2]: Ok

**Review by nb-ohad — DISMISSED** [review]

**Review by ezio-auditore — APPROVED** [review]

**Review by nb-ohad — APPROVED** [review]

**Pull request merged**